Library/Homebrew/requirements/gpg_requirement.rb

```diff
@@ -0,0 +1,29 @@
+require "requirement"
+
+class GPGRequirement < Requirement
+  fatal true
+  default_formula "gnupg2"
+
+  satisfy(:build_env => false) { gpg2 || gpg1 }
+
+  # Homebrew installs GnuPG 1.x as `gpg` but this may change in future.
+  # MacGPG2/GPGTools installs GnuPG 2.0.x as a vanilla `gpg` symlink
+  # pointing to `gpg2`. Ensure we're not using unexpected gpg version.
+  def gpg1
+    which_all("gpg").detect do |gpg|
+      gpg_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_version
+      Version.new(gpg_version.to_s) == Version.new("1.4")
+    end
+  end
+
+  def gpg2
+    which_all("gpg2").detect do |gpg2|
+      gpg2_version = Utils.popen_read(gpg2, "--version")[/\d\.\d/, 0]
+      next unless gpg2_version
+      # For now, to give upstreams time to adjust, only support 2.0.x
+      # rather than the 2.1.x "modern" series.
+      Version.new(gpg2_version.to_s) == Version.new("2.0")
+    end
+  end
```

Comment:
> That means gpg 2.1 alone works just fine, it's just some formulae who depend on it may break.

Yeah, this is a fairly big deal given that Requirements exist to satisfy formulae...

According to GA 325 people have downloaded `gnupg21` from Versions since GA went live, Bintray says half that.

Statistically there's not a huge rush to switch over to `gnupg21`, it complicates the requirement for us to do so, and you're effectively volunteering me to go back & extensively test every `gpg`-using core formulae to determine an upper limit on what they'll support, which frankly as much as I use `gpg` I'm not qualified to make a determination on that much of an intimate level of how every upstream is using `gpg` internally.